Alignment/CommonAlignmentProducer/plugins/PCLTrackerAlProducer.cc

```diff
@@ -0,0 +1,1247 @@
+{
+  createAlignmentAlgorithm(config);
+  createCalibrations      (config);
+  createMonitors          (config);
+}
+
+//_____________________________________________________________________________
+PCLTrackerAlProducer
+::~PCLTrackerAlProducer()
+{
+  delete theAlignmentAlgo;
+
+  for (auto iCal  = theCalibrations.begin();
+            iCal != theCalibrations.end();
+          ++iCal) {
+    delete *iCal;
+  }
+
+  // TODO: Delete monitors as well?
+
+  delete theAlignmentParameterStore;
+  delete theTrackerAlignables;
+  delete theMuonAlignables;
+  delete theExtraAlignables;
+  delete globalPositions_;
+}
+
+
+
+//=============================================================================
+//===   INTERFACE IMPLEMENTATION                                            ===
+//=============================================================================
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::beginJob()
+{
+  nevent_ = 0;
+
+  for (auto iCal  = theCalibrations.begin();
+            iCal != theCalibrations.end();
+          ++iCal) {
+    (*iCal)->beginOfJob(theTrackerAlignables,
+                        theMuonAlignables,
+                        theExtraAlignables);
+  }
+
+  for (auto monitor  = theMonitors.begin();
+            monitor != theMonitors.end();
+          ++monitor) {
+     (*monitor)->beginOfJob(theTrackerAlignables,
+                            theMuonAlignables,
+                            theAlignmentParameterStore);
+  }
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::endJob()
+{
+  finish();
+
+  for (auto monitor  = theMonitors.begin();
+            monitor != theMonitors.end();
+          ++monitor) {
+    (*monitor)->endOfJob();
+  }
+
+  for (auto iCal  = theCalibrations.begin();
+            iCal != theCalibrations.end();
+          ++iCal) {
+    (*iCal)->endOfJob();
+  }
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::beginRun(const edm::Run& run, const edm::EventSetup& setup)
+{
+  if (setupChanged(setup)) {
+    edm::LogInfo("Alignment") << "@SUB=PCLTrackerAlProducer::beginRun"
+                              << "EventSetup-Record changed.";
+    initAlignmentAlgorithm(setup);
+  }
+
+  // Do not forward edm::Run
+  theAlignmentAlgo->beginRun(setup);
+
+  if (setupChanged(setup)) {
+    initAlignmentAlgorithm(setup);
+  }
+
+
+}
+
+//_____________________________________________________________________________
+void PCLTrackerAlProducer
+::endRun(const edm::Run& run, const edm::EventSetup& setup)
+{
+  // TODO: Either MP nor HIP is implementing the endRun() method... so this
+  //       seems to be useless?
+
+  if (tkLasBeamTag_.encode().size()) {
+    edm::Handle<TkFittedLasBeamCollection> lasBeams;
+    edm::Handle<TsosVectorCollection> tsoses;
+    run.getByLabel(tkLasBeamTag_, lasBeams);
```

Comment:
@cerminar @davidlange6 Sorry, there is a type in the TODO comment, should be 'neither'... the method call in line 198 (and 203) is useless, because neither MP nor HIP implements this method, so also this method seems to be useless.